### PR TITLE
Allow re-registration for events without creating a new record.

### PR DESCRIPTION
**Pull request by MDevster**

### CODE REVIEW

**Comment by fritzmg** [comment]
Hm, not sure I understand the use case? So members should have the possibility to override their previous registration with new values?

File: src/EventListener/EventRegistrationFormListener.php [hunk 1]
```diff
@@ -52,9 +52,19 @@ public function __invoke(array &$submittedData, array $formData, ?array $files,
 
         $member = $this->getMember();
 
-        $registration = new EventRegistrationModel();
+        // check if registration exists
+        $registration = EventRegistrationModel::findOneBy(['pid = ?', 'member = ?'], [(int) $event->id, (int) $member->id]);
+
+        if (null !== $registration) {
+            $registration->cancelled = 0;
```
Comment (fritzmg) [inline src/EventListener/EventRegistrationFormListener.php:59]:
```suggestion
            $registration->cancelled = 0;
            $registration->confirmed = 0;
```

I think you also need to set `confirmed` to false as the registration would potentially need to be confirmed again.

Comment (MDevster) [reply src/EventListener/EventRegistrationFormListener.php:59]: Correct!

**Comment by MDevster** [comment]
**Use case**
A member canceled the registration for an event. Now he wants to register again, either because the event has been postponed or because he has time at short notice.
When registering again, a new record was always created, although the user was logged in. This caused the oldest record to be used when checking for canceled or confirmed in the templates and hooks.

Now logged in members can register only once per event and the existing record is updated.

**Comment by fritzmg** [comment]
> This caused the oldest record to be used when checking for canceled or confirmed in the templates and hooks.

What do you mean by that? This sounds like something that should be fixed instead.

I think it's also a valid use case to have multiple registrations for the same member (regardless of whether they are confirmed or cancelled etc.).

**Comment by MDevster** [comment]
You are right of course, other users may want to allow multiple registrations. The problem then lies with the other modules, e.g.  https://github.com/inspiredminds/contao-event-registration/blob/main/src/Controller/FrontendModule/EventRegistrationCancelController.php#L65

I would then have to get a list of my registrations and decide for each one individually if I want to cancel or reactivate it.

**Comment by MDevster** [comment]
Same her https://github.com/inspiredminds/contao-event-registration/blob/main/src/Controller/FrontendModule/EventRegistrationConfirmController.php#L66

**Comment by fritzmg** [comment]
> You are right of course, other users may want to allow multiple registrations. The problem then lies with the other modules, e.g. https://github.com/inspiredminds/contao-event-registration/blob/main/src/Controller/FrontendModule/EventRegistrationCancelController.php#L65
>
> I would then have to get a list of my registrations and decide for each one individually if I want to cancel or reactivate it.

> Same her https://github.com/inspiredminds/contao-event-registration/blob/main/src/Controller/FrontendModule/EventRegistrationConfirmController.php#L66

I don't see any problem there. If you do multiple registrations you get multiple emails, each containing a cancel and confirm link for that particular registration. If you would like to cancel or confirm all your registration you would need to cancel or confirm each registration individually.

Overall I don't think it is correct to be able to overwrite an existing registration like that. Besides, it would only work for members.

**Comment by MDevster** [comment]
Ok yes, via the email it works, but the member should be able to unsubscribe via the details page. So it was the easiest to provide a link like this
`<a href="{{env::request}}?action=cancel&uuid=<?= $this->uuid ?>">Unregister from the event</a>`
But then this was always executed only for the first event. I also just took over the project, I'll look at the context again and possibly make a new PR later. You can close that one for now.

**Comment by fritzmg** [comment]
That would be a new feature which should be implemented as a separate module. There is already an existing module that shows registrations for the current event. May be that can be re used with a new setting where you can define that only events for the current member should be shown. And each entry could then have a confirm and cancel link (depending on the configuration of the event itself).

**Review by MDevster — COMMENTED** [review]
can be merged

**Pull request closed without merging**